src/components/structures/MatrixChat.js

```diff
@@ -304,6 +305,9 @@ module.exports = React.createClass({
                 });
                 break;
             case 'view_room':
+                // by default we autoPeek rooms, unless we were called explicitly with
+                // autoPeek=false by something like RoomDirectory who has already peeked
+                this.setState({ autoPeek : payload.auto_peek === false ? false : true });
```

Comment:
i want:

```
if (payload.auto_peek === undefined) {
    this.setState({ autoPeek : true }); // default to true if not specified
}
else {
    this.setState({ autoPeek: payload.auto_peek });
}
```

which is what the ternary is trying to do more succinctly.  Which of us is on the wrong page? :D